images/krte/wrapper.sh

```diff
@@ -13,66 +13,72 @@
+printf '%0.s=' {1..80}; echo
+echo "wrapper.sh] Performing setup ..."
 
 # Check if the job has opted-in to bazel remote caching and if so generate 
 # .bazelrc entries pointing to the remote cache
 export BAZEL_REMOTE_CACHE_ENABLED=${BAZEL_REMOTE_CACHE_ENABLED:-false}
 if [[ "${BAZEL_REMOTE_CACHE_ENABLED}" == "true" ]]; then
-    echo "Bazel remote cache is enabled, generating .bazelrcs ..."
-    /usr/local/bin/create_bazel_cache_rcs.sh
+  echo "wrapper.sh] [SETUP] Bazel remote cache is enabled, generating .bazelrcs ..."
+  /usr/local/bin/create_bazel_cache_rcs.sh
+  echo "wrapper.sh] [SETUP] Done setting up .bazelrcs"
 fi
 
 
-# terminate any remaining containers
+# terminates any remaining containers
 cleanup_dind() {
-    docker ps -aq | xargs docker rm -f
+  docker ps -aq | xargs -r docker rm -f || true
 }
 
 # optionally enable ipv6 docker
 export DOCKER_IN_DOCKER_IPV6_ENABLED=${DOCKER_IN_DOCKER_IPV6_ENABLED:-false}
 if [[ "${DOCKER_IN_DOCKER_IPV6_ENABLED}" == "true" ]]; then
-    echo "Enabling IPV6 for Docker."
-    # configure the daemon with ipv6
-    mkdir -p /etc/docker/
-    cat <<EOF >/etc/docker/daemon.json
+  echo "wrapper.sh] [SETUP] Enabling IPv6 in Docker config ..."
+  # configure the daemon with ipv6
+  mkdir -p /etc/docker/
+  cat <<EOF >/etc/docker/daemon.json
 {
   "ipv6": true,
   "fixed-cidr-v6": "fc00:db8:1::/64"
 }
 EOF
-    # enable ipv6
-    sysctl net.ipv6.conf.all.disable_ipv6=0
-    sysctl net.ipv6.conf.all.forwarding=1
-    # enable ipv6 iptables
-    modprobe -v ip6table_nat
+  # enable ipv6
+  sysctl net.ipv6.conf.all.disable_ipv6=0
+  sysctl net.ipv6.conf.all.forwarding=1
+  # enable ipv6 iptables
+  modprobe -v ip6table_nat
+  echo "wrapper.sh] [SETUP] Done enabling IPv6 in Docker config."
 fi
 
 # Check if the job has opted-in to docker-in-docker availability.
 export DOCKER_IN_DOCKER_ENABLED=${DOCKER_IN_DOCKER_ENABLED:-false}
 if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
-    echo "Docker in Docker enabled, initializing..."
-    printf '=%.0s' {1..80}; echo
-    # If we have opted in to docker in docker, start the docker daemon,
-    service docker start
-    # the service can be started but the docker socket not ready, wait for ready
-    WAIT_N=0
-    MAX_WAIT=5
-    while true; do
-        # docker ps -q should only work if the daemon is ready
-        docker ps -q > /dev/null 2>&1 && break
-        if [[ ${WAIT_N} -lt ${MAX_WAIT} ]]; then
-            WAIT_N=$((WAIT_N+1))
-            echo "Waiting for docker to be ready, sleeping for ${WAIT_N} seconds."
-            sleep ${WAIT_N}
-        else
-            echo "Reached maximum attempts, not waiting any longer..."
-            break
-        fi
-    done
-    cleanup_dind
-    printf '=%.0s' {1..80}; echo
-    echo "Done setting up docker in docker."
+  echo "wrapper.sh] [SETUP] Docker in Docker enabled, initializing ..."
+  # If we have opted in to docker in docker, start the docker daemon,
+  service docker start
```

Comment:
is not using systemd?

Reply:
no, systemd + docker in docker is a pain. unnecessary for this image. this isn't new.